core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java

```diff
@@ -389,6 +261,51 @@ public static Map<Long, String> getTableObjectsFullName(List<Long> tableIds) {
     return tableIdAndNameMap;
   }
 
+  /**
+   * Retrieves a map of column object IDs to their full names.
+   *
+   * @param columnsIds A list of column object IDs to fetch names for.
+   * @return A Map where the key is the column ID and the value is the column full name. The map may
+   *     contain null values for the names if its parent object is deleted. Returns an empty map if
+   *     no column objects are found for the given IDs. {@code @example} value of table full name:
+   *     "catalog1.schema1.table1.column1"
+   */
+  public static Map<Long, String> getColumnObjectsFullName(List<Long> columnsIds) {
+    List<ColumnPO> columnPOs =
+        SessionUtils.getWithoutCommit(
+            TableColumnMapper.class, mapper -> mapper.listColumnPOsByTableIds(columnsIds));
+
+    if (columnPOs == null || columnPOs.isEmpty()) {
+      return new HashMap<>();
+    }
+
+    List<Long> tableIds = columnPOs.stream().map(ColumnPO::getTableId).collect(Collectors.toList());
+    Map<Long, String> tableIdAndNameMap = getTableObjectsFullName(tableIds);
+
+    HashMap<Long, String> columnIdAndNameMap = new HashMap<>();
+
+    columnPOs.forEach(
+        columnPO -> {
+          // since the table can be deleted, we need to check the null value,
+          // and when the table is deleted, we will set fullName of column to
+          // null
+          String tableName = tableIdAndNameMap.getOrDefault(columnPO.getTableId(), null);
+          if (tableName == null) {
+            LOG.warn(
+                "The table '{}' of column '{}' may be deleted",
+                columnPO.getTableId(),
+                columnPO.getColumnId());
+            columnIdAndNameMap.put(columnPO.getColumnId(), null);
+            return;
```

Comment:
What will be happened if table name is null? Shall we throw an exception here?

Reply:
According to the comment and code logic, the table can be deleted when when query the columns, we should ingore it. please see:

https://github.com/apache/gravitino/blob/2668aaf2bd01120318fb80f8dd010364cd330d0d/core/src/main/java/org/apache/gravitino/storage/relational/service/TagMetaService.java#L238-L242